BibBuddy/app/src/main/java/de/bibbuddy/BookFragment.java

```diff
@@ -669,4 +638,20 @@ public void onLongItemClicked(int position, BookItem bookItem, View v) {
     }
   }
 
+  private void shareShelfBibIntent() {
+
+    Uri contentUri = exportBibTex.writeTemporaryBibFile(context,
+        exportBibTex.getBibDataFromShelf(shelfId, bookDao, noteDao));
+
+    Intent shareShelfIntent =
+        ShareCompat.IntentBuilder.from(getActivity())
+            .setStream(contentUri)
+            .setType("text/*")
+            .getIntent()
+            .addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
+
+    startActivity(Intent.createChooser(shareShelfIntent, "SEND"));
```

Comment:
Maybe use a variable for "SEND" because you also use it in LibraryFragment.

Reply:
> Is it possible to create an new class to open the same intent (LibraryFragment, BookFragment) with different bib data.

As far as I tried and researched -> no, because you cannot call the startActivity() method in our case.

Reply:
If so, then I need to create a new class with 2 constants, because the "text/*" and "SEND" variables are specific only for the share intent. I think we do not need to create a new class with only two constants.